proposal/prefixed-parent-selector.md

```diff
@@ -70,28 +68,90 @@ Another [example](https://github.com/sass/sass/issues/1425#issuecomment-21207790
-[Comment #1](https://github.com/sass/sass/issues/1425#issuecomment-55462526)
-[Comment #2](https://github.com/sass/sass/issues/1425#issuecomment-123913176)
-[Comment #3](https://github.com/sass/sass/issues/1425#issuecomment-123916395)
+Note that this syntax is supported by Less and Stylus, and people have run into the lack of support for this syntax when trying to switch over to Sass:
 
-There has been an attempt to create a proposal for it, but it fell short in addressing review comments and was closed soon after.
+* [Comment #1](https://github.com/sass/sass/issues/1425#issuecomment-55462526)
+* [Comment #2](https://github.com/sass/sass/issues/1425#issuecomment-123913176)
+* [Comment #3](https://github.com/sass/sass/issues/1425#issuecomment-123916395)
+
+There has been an attempt to create a proposal for it, but it fell short in addressing review comments and was closed soon after:
 [Previous Proposal](https://github.com/sass/sass/pull/2723)
 
+## Syntax
+(In terms of [Selectors Level 4 Grammar](https://www.w3.org/TR/selectors-4/#grammar))
+
+```
+SelectorList         ::= ComplexSelectorList#
+ComplexSelectorList  ::= ComplexSelector#
+CompoundSelectorList ::= CompoundSelector#
+SimpleSelectorList   ::= SimpleSelector#
+RelativeSelectorList ::= RelativeSelector#
+ComplexSelector      ::= CompoundSelector [ <combinator>? CompoundSelector ]*
+RelativeSelector     ::= <combinator>? ComplexSelector
+CompoundSelector     ::= ParentSelector? <compound-selector>
+                         | <compound-selector> ParentSelector
+                         | ParentSelector
+SimpleSelector       ::= ParentSelector? <simple-selector>
+                         | <simple-selector> ParentSelector
+                         | ParentSelector
+ParentSelector       ::= '&'
+```
+
+Note that it is not necessary to support the following syntax:
+```
+.custom-effect {
+  p&:focus {
+    // styles for <p> when focused
+  }
+}
+```
+
+Because the same effect can be achieved with one extra level of nesting:
+```
+.custom-effect {
+  p& {
+    &:focus {
+      // styles for <p> when focused
+    }
+  }
+}
+```
+
 ## Semantics
 
-The [Style Rules Semantics](https://github.com/sass/sass/blob/master/spec/style-rules.md#semantics) describes the parent selector behavior as follows:
+The following modifies the [Style Rules Semantics](https://github.com/sass/sass/blob/master/spec/style-rules.md#semantics):
+
+To execute a style rule `rule`:
+
+* Let `selector` be the result of evaluating all interpolation in `rule`'s
+  selector and parsing the result as a selector list.
+
+* If there is a current style rule:
+
+  * If `selector` contains one or more `ParentSelector`:
+
+      * If multiple `ParentSelector` appear in the a single `SimpleSelector` or `CompoundSelector`,
+        throw an error.
+
+      * Otherwise, replace each `ParentSelector` with the current style rule's
+        selector and set `selector` to the result.
```

Comment:
I think this still needs some more detail to describe how to handle `ParentSelector`s with prefixes and suffixes. For example, it should be detailed enough to explain that `* { foo& { /* ... */ } }` is an error and how `.bar { foo& { /* ... */ } }` works differently than `bar { foo& { /* ... */ } }`.